go-controller/pkg/ovn/base_network_controller.go

```diff
@@ -696,12 +700,28 @@ func (bnc *BaseNetworkController) doesNetworkRequireIPAM() bool {
 	return util.DoesNetworkRequireIPAM(bnc.NetInfo)
 }
 
-func (bnc *BaseNetworkController) getPodNADNames(pod *kapi.Pod) []string {
+func (bnc *BaseNetworkController) getPodNADNames(pod *kapi.Pod) ([]string, error) {
 	if !bnc.IsSecondary() {
-		return []string{types.DefaultNetworkName}
+		return []string{types.DefaultNetworkName}, nil
+	}
+	if bnc.IsPrimaryNetwork() {
+		// We couldn't rely on bnc.NetInfo because this may span across different namespaces.
```

Comment:
You could also comment here that `PodNadNames` is based on the network selection element so it wouldn't work. All this implementation could be in `util.PodNadNames` I think

Reply:
added comment and i tried to move this logic into util.PodNadNames, but it requires nadLister nadlister.NetworkAttachmentDefinitionLister, so needed intrusive changes across all the calling methods. do you still this to be part of this PR or can it be done separately ?

Reply:
But the lister is already being passed to the global function `util.GetActiveNetworkForNamespace`